### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by segasai**
Several fixes addressing #21, #19, #18 
Also run the code through yapf formatter.

I.e now when specifying invalid mmin for kroupa a proper error is raised
```python
In [5]: cl = imf.make_cluster(1000,massfunc='kroupa',mmin=1)
ValueError: Power law break-points must be monotonic
```
also the global imf objects are not modified since get_massfunc() returns new instances

Also mmin is supported now for chabrier
```python
In [13]: imf.make_cluster(1000,massfunc='chabrier',mmin=10).mean()
Total cluster mass is 1002.2 (limit was 1000)
Out[13]: 16.70338008453732

In [14]: imf.make_cluster(1000,massfunc='chabrier',mmin=1).mean()
Total cluster mass is 1000.38 (limit was 1000)
Out[14]: 2.2684320853833024
```

### CODE REVIEW

File: imf/imf.py [hunk 1]
```diff
@@ -244,9 +256,6 @@ def __call__(self, x, integral_form=False, **kw):
             return self.distr.pdf(x)
 
 
-chabrier2005 = Chabrier2005()
```
Comment (keflavich) [inline imf/imf.py:247]: Don't remove this, or the others.

File: imf/imf.py [hunk 2]
```diff
@@ -183,40 +179,56 @@ def m_integrate(self, mlow, mhigh, numerical=False, **kwargs):
...
     def __call__(self, mass, integral_form=False, **kw):
         if integral_form:
-            return self.distr.cdf(mass)*self.multiplier
+            return self.distr.cdf(mass) * self.multiplier
         else:
-            return self.distr.pdf(mass)*self.multiplier
-
+            return self.distr.pdf(mass) * self.multiplier
 
-chabrier = Chabrier()
```
Comment (keflavich) [inline imf/imf.py:215]: Don't remove this

File: imf/imf.py [hunk 3]
```diff
@@ -183,40 +179,56 @@ def m_integrate(self, mlow, mhigh, numerical=False, **kwargs):
...
-                                                   -self.p3+1],
-                                                  [self.mmin, self.break1,
-                                                   self.break2, self.mmax])
-            ratio = distr1.pdf(self.break1)/self.distr.pdf(self.break1)/self.break1
-            return ((distr1.cdf(mhigh)-distr1.cdf(mlow))/ratio, 0)
-
-
-
-
-kroupa = Kroupa()
```
Comment (keflavich) [inline imf/imf.py:196]: Don't remove this

File: imf/imf.py [hunk 4]
```diff
@@ -98,16 +96,16 @@ def __call__(self, m, integral_form=False):
             return self.distr.cdf(m) * self.normfactor
 
 
-# three codes for dn/dlog(m)
-salpeter = Salpeter()
```
Comment (keflavich) [inline imf/imf.py:102]: Don't remove this

**Review by keflavich — CHANGES_REQUESTED** [review]
I'm 👍 on most of these changes, but the globally-defined instances of the classes should not be removed - they are there as convenience functions and I want to keep them.  This request is compatible with the change to `get_massfunc`,  right?

Big :+1: on the `yapf` reformatting; I wrote much of this code before I had a good linter installed.

**Comment by segasai** [comment]
I didn't remove the global objects -- I've groupped them. (see lines ~ 300).
The reason to separate the definition of the classes from the creation of global objects.
And get_massfunc works after the change (if I understood your question)

**Comment by keflavich** [comment]
Sorry, that's my bad!  I was looking for exactly
https://github.com/keflavich/imf/pull/26/files#diff-1e13df7a25dad73166b0f0e1e9909d2829360f05f1077e4a0d60d64e24bfc7c2R340-R344
but didn't find them.  That's the problem with including style fixes in with others.

**Review by keflavich — APPROVED** [review]

**Pull request merged**